### PR TITLE
CSSOM View: Expand on scrollIntoView test

### PR DESCRIPTION
**Pull request by zcorpan**
Follows w3c/csswg-drafts#1505

### CODE REVIEW

**Comment by zcorpan** [comment]
cc @sunyunjia

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/243278483)

# Firefox (nightly) #
Testing web-platform-tests at revision 1ca1e676dcd995ca693bebfb7c095a977830d89c
Using browser at version BuildID 20170612100241; SourceStamp 27cad9749cddf68e11fdd4e5d73dad84a8f8cf23
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6253/cssom-view/scrollintoview.html">/cssom-view/scrollintoview.html</a></summary>


|                                     Subtest                                      | Results |                                                     Messages                                                    |
|----------------------------------------------------------------------------------|---------|-----------------------------------------------------------------------------------------------------------------|
|                                                                                  | OK      |                                                                                                                 |
| `scrollIntoView() starting at left,top`                                          | PASS    |                                                                                                                 |
| `scrollIntoView() starting at left,bottom`                                       | PASS    |                                                                                                                 |
| `scrollIntoView() starting at right,top`                                         | PASS    |                                                                                                                 |
| `scrollIntoView() starting at right,bottom`                                      | PASS    |                                                                                                                 |
| `scrollIntoView(true) starting at left,top`                                      | PASS    |                                                                                                                 |
| `scrollIntoView(true) starting at left,bottom`                                   | PASS    |                                                                                                                 |
| `scrollIntoView(true) starting at right,top`                                     | PASS    |                                                                                                                 |
| `scrollIntoView(true) starting at right,bottom`                                  | PASS    |                                                                                                                 |
| `scrollIntoView(false) starting at left,top`                                     | PASS    |                                                                                                                 |
| `scrollIntoView(false) starting at left,bottom`                                  | PASS    |                                                                                                                 |
| `scrollIntoView(false) starting at right,top`                                    | PASS    |                                                                                                                 |
| `scrollIntoView(false) starting at right,bottom`                                 | PASS    |                                                                                                                 |
| `scrollIntoView(undefined) starting at left,top`                                 | PASS    |                                                                                                                 |
| `scrollIntoView(undefined) starting at left,bottom`                              | PASS    |                                                                                                                 |
| `scrollIntoView(undefined) starting at right,top`                                | PASS    |                                                                                                                 |
| `scrollIntoView(undefined) starting at right,bottom`                             | PASS    |                                                                                                                 |
| `scrollIntoView({}) starting at left,top`                                        | FAIL    | `assert_approx_equals: scrollX expected 3639.5 +/- 0.5 but got 3279`                                            |
| `scrollIntoView({}) starting at left,bottom`                                     | FAIL    | `assert_approx_equals: scrollX expected 3639.5 +/- 0.5 but got 3279`                                            |
| `scrollIntoView({}) starting at right,top`                                       | FAIL    | `assert_approx_equals: scrollX expected 3639.5 +/- 0.5 but got 4000`                                            |
| `scrollIntoView({}) starting at right,bottom`                                    | FAIL    | `assert_approx_equals: scrollX expected 3639.5 +/- 0.5 but got 4000`                                            |
| `scrollIntoView({block: "center", inline: "center"}) starting at left,top`       | FAIL    | `'block' member of ScrollIntoViewOptions 'center' is not a valid value for enumeration ScrollLogicalPosition.`  |
| `scrollIntoView({block: "center", inline: "center"}) starting at left,bottom`    | FAIL    | `'block' member of ScrollIntoViewOptions 'center' is not a valid value for enumeration ScrollLogicalPosition.`  |
| `scrollIntoView({block: "center", inline: "center"}) starting at right,top`      | FAIL    | `'block' member of ScrollIntoViewOptions 'center' is not a valid value for enumeration ScrollLogicalPosition.`  |
| `scrollIntoView({block: "center", inline: "center"}) starting at right,bottom`   | FAIL    | `'block' member of ScrollIntoViewOptions 'center' is not a valid value for enumeration ScrollLogicalPosition.`  |
| `scrollIntoView({block: "start", inline: "start"}) starting at left,top`         | FAIL    | `assert_approx_equals: scrollX expected 4000 +/- 0.5 but got 3279`                                              |
| `scrollIntoView({block: "start", inline: "start"}) starting at left,bottom`      | FAIL    | `assert_approx_equals: scrollX expected 4000 +/- 0.5 but got 3279`                                              |
| `scrollIntoView({block: "start", inline: "start"}) starting at right,top`        | PASS    |                                                                                                                 |
| `scrollIntoView({block: "start", inline: "start"}) starting at right,bottom`     | PASS    |                                                                                                                 |
| `scrollIntoView({block: "end", inline: "end"}) starting at left,top`             | PASS    |                                                                                                                 |
| `scrollIntoView({block: "end", inline: "end"}) starting at left,bottom`          | PASS    |                                                                                                                 |
| `scrollIntoView({block: "end", inline: "end"}) starting at right,top`            | FAIL    | `assert_approx_equals: scrollX expected 3279 +/- 0.5 but got 4000`                                              |
| `scrollIntoView({block: "end", inline: "end"}) starting at right,bottom`         | FAIL    | `assert_approx_equals: scrollX expected 3279 +/- 0.5 but got 4000`                                              |
| `scrollIntoView({block: "nearest", inline: "nearest"}) starting at left,top`     | FAIL    | `'block' member of ScrollIntoViewOptions 'nearest' is not a valid value for enumeration ScrollLogicalPosition.` |
| `scrollIntoView({block: "nearest", inline: "nearest"}) starting at left,bottom`  | FAIL    | `'block' member of ScrollIntoViewOptions 'nearest' is not a valid value for enumeration ScrollLogicalPosition.` |
| `scrollIntoView({block: "nearest", inline: "nearest"}) starting at right,top`    | FAIL    | `'block' member of ScrollIntoViewOptions 'nearest' is not a valid value for enumeration ScrollLogicalPosition.` |
| `scrollIntoView({block: "nearest", inline: "nearest"}) starting at right,bottom` | FAIL    | `'block' member of ScrollIntoViewOptions 'nearest' is not a valid value for enumeration ScrollLogicalPosition.` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/243278484)

# Sauce (safari) #
Testing web-platform-tests at revision 1ca1e676dcd995ca693bebfb7c095a977830d89c
Using browser at version 10.0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/6253/cssom-view/scrollintoview.html">/cssom-view/scrollintoview.html</a></summary>


|                                     Subtest                                      | Results |                              Messages                              |
|----------------------------------------------------------------------------------|---------|--------------------------------------------------------------------|
|                                                                                  | OK      |                                                                    |
| `scrollIntoView() starting at left,top`                                          | PASS    |                                                                    |
| `scrollIntoView() starting at left,bottom`                                       | PASS    |                                                                    |
| `scrollIntoView() starting at right,top`                                         | PASS    |                                                                    |
| `scrollIntoView() starting at right,bottom`                                      | PASS    |                                                                    |
| `scrollIntoView(true) starting at left,top`                                      | PASS    |                                                                    |
| `scrollIntoView(true) starting at left,bottom`                                   | PASS    |                                                                    |
| `scrollIntoView(true) starting at right,top`                                     | PASS    |                                                                    |
| `scrollIntoView(true) starting at right,bottom`                                  | PASS    |                                                                    |
| `scrollIntoView(false) starting at left,top`                                     | PASS    |                                                                    |
| `scrollIntoView(false) starting at left,bottom`                                  | PASS    |                                                                    |
| `scrollIntoView(false) starting at right,top`                                    | PASS    |                                                                    |
| `scrollIntoView(false) starting at right,bottom`                                 | PASS    |                                                                    |
| `scrollIntoView(undefined) starting at left,top`                                 | PASS    |                                                                    |
| `scrollIntoView(undefined) starting at left,bottom`                              | PASS    |                                                                    |
| `scrollIntoView(undefined) starting at right,top`                                | PASS    |                                                                    |
| `scrollIntoView(undefined) starting at right,bottom`                             | PASS    |                                                                    |
| `scrollIntoView({}) starting at left,top`                                        | FAIL    | `assert_approx_equals: scrollX expected 3700 +/- 0.5 but got 3400` |
| `scrollIntoView({}) starting at left,bottom`                                     | FAIL    | `assert_approx_equals: scrollX expected 3700 +/- 0.5 but got 3400` |
| `scrollIntoView({}) starting at right,top`                                       | FAIL    | `assert_approx_equals: scrollX expected 3700 +/- 0.5 but got 4000` |
| `scrollIntoView({}) starting at right,bottom`                                    | FAIL    | `assert_approx_equals: scrollX expected 3700 +/- 0.5 but got 4000` |
| `scrollIntoView({block: "center", inline: "center"}) starting at left,top`       | FAIL    | `assert_approx_equals: scrollX expected 3700 +/- 0.5 but got 3400` |
| `scrollIntoView({block: "center", inline: "center"}) starting at left,bottom`    | FAIL    | `assert_approx_equals: scrollX expected 3700 +/- 0.5 but got 3400` |
| `scrollIntoView({block: "center", inline: "center"}) starting at right,top`      | FAIL    | `assert_approx_equals: scrollX expected 3700 +/- 0.5 but got 4000` |
| `scrollIntoView({block: "center", inline: "center"}) starting at right,bottom`   | FAIL    | `assert_approx_equals: scrollX expected 3700 +/- 0.5 but got 4000` |
| `scrollIntoView({block: "start", inline: "start"}) starting at left,top`         | FAIL    | `assert_approx_equals: scrollX expected 4000 +/- 0.5 but got 3400` |
| `scrollIntoView({block: "start", inline: "start"}) starting at left,bottom`      | FAIL    | `assert_approx_equals: scrollX expected 4000 +/- 0.5 but got 3400` |
| `scrollIntoView({block: "start", inline: "start"}) starting at right,top`        | PASS    |                                                                    |
| `scrollIntoView({block: "start", inline: "start"}) starting at right,bottom`     | PASS    |                                                                    |
| `scrollIntoView({block: "end", inline: "end"}) starting at left,top`             | FAIL    | `assert_approx_equals: scrollY expected 3638 +/- 0.5 but got 4000` |
| `scrollIntoView({block: "end", inline: "end"}) starting at left,bottom`          | FAIL    | `assert_approx_equals: scrollY expected 3638 +/- 0.5 but got 4000` |
| `scrollIntoView({block: "end", inline: "end"}) starting at right,top`            | FAIL    | `assert_approx_equals: scrollX expected 3400 +/- 0.5 but got 4000` |
| `scrollIntoView({block: "end", inline: "end"}) starting at right,bottom`         | FAIL    | `assert_approx_equals: scrollX expected 3400 +/- 0.5 but got 4000` |
| `scrollIntoView({block: "nearest", inline: "nearest"}) starting at left,top`     | FAIL    | `assert_approx_equals: scrollY expected 3638 +/- 0.5 but got 4000` |
| `scrollIntoView({block: "nearest", inline: "nearest"}) starting at left,bottom`  | PASS    |                                                                    |
| `scrollIntoView({block: "nearest", inline: "nearest"}) starting at right,top`    | FAIL    | `assert_approx_equals: scrollY expected 3638 +/- 0.5 but got 4000` |
| `scrollIntoView({block: "nearest", inline: "nearest"}) starting at right,bottom` | PASS    |                                                                    |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/243278490)

# Chrome (unstable) #
Testing web-platform-tests at revision 1ca1e676dcd995ca693bebfb7c095a977830d89c

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/243278491)

# Sauce (MicrosoftEdge) #
Testing web-platform-tests at revision 1ca1e676dcd995ca693bebfb7c095a977830d89c
Using browser at version 14.14393
Starting 10 test iterations

**Comment by sunyunjia** [comment]
Hi Simon,
Does this mean we expect the element to be scrolled to "top, nearest" when we pass in "undefined" or "null"?

**Comment by zcorpan** [comment]
undefined, yes, see the spec change. Null should be like `{}` I believe, I missed adding a test for that.

**Comment by zcorpan** [comment]
Added a test for null

**Comment by w3c-bots** [comment]
# Build PENDING

Started: None
Finished: None





View more information about this build on:

- [WPT PR Status](http://pulls.web-platform-tests.org/build/18323)
- [TravisCI](https://travis-ci.org/w3c/web-platform-tests/builds/273258685)

File: cssom-view/scrollintoview.html [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+body.running { margin: 0; padding: 4000px; overflow: hidden }
+body.running #testDiv {
+  width: 200px;
+  height: 200px;
+  background-color: lightgreen;
+}
+</style>
+<body class=running>
+<div id=testDiv></div>
+<div id="log"></div>
```
Comment (rune-opera) [inline cssom-view/scrollintoview.html]: Won't this be created by the testharness? Does the automatic creating affect the testing here?

Comment (zcorpan) [reply cssom-view/scrollintoview.html]: testharness.js creates it if it doesn't already exist, but it's OK to have it there. It doesn't affect the test either way.

**Review by ghost — APPROVED** [review]

**Pull request merged**